addons/stock/tests/common.py

```diff
@@ -179,6 +213,84 @@ def setUpClass(cls):
     def url_extract_rec_id_and_model(self, url):
         # Extract model and record ID
         action_match = re.findall(r'action-([^/]+)', url)
-        model_name = self.env.ref(action_match[0]).res_model
+        model_name = self.env.ref(action_match[0]).sudo().res_model
         rec_id = re.findall(r'/(\d+)$', url)[0]
         return rec_id, model_name
+
+    @classmethod
+    def setup_independent_user(cls):
+        return new_test_user(
+            cls.env,
+            name='Test Stock User',
+            login='stock',
+            password='stockpass',
+            email='stock@test.com',
+            groups_id=[cls.env.ref('stock.group_stock_manager').id],
+        )
```

Comment:
I know I did recommend using that existing API, but honesty, considering the sudo you had to add here and there, it confirms that I don't like this approach at all.
IMHO, we shouldn't have to modify existing working tests when introducing a new one, unless it's to simplify, speed them up or add new features to the API.

Also, what annoys me the most with this approach is that the environment of the setup variables is not constant at all.
```python
self.user_stock_user.env.su # True
self.company_data['default_sale_team'].env.su # True
self.tax_armageddon.env.su # True
self.env.su # False
self.company.env.su # True
self.warehouse_1.env.su # False
```

When creating a new test, if you use existing variables from common, you might do some things in sudo or not, without being aware of it.  This is confusing and unclear, which is not what I'd expect from a common.
I believe it's way clearer if all setup variables are created and defined with admin rights, and the environment is 'optionally' changed to the stock user/manager in a `setUp` override instead, after the classes setup.

I know that you managed to notice a lot of access rights issues that way, but I'm not sure it should be come the standard behavior, at least not the way it is right now.